### PR TITLE
Bugfix/remove unused functions4

### PR DESCRIPTION
**Pull request by pranavtifr**
I'll rebase after merging #1079

### CODE REVIEW

File: common/city.cpp [hunk 1]
```diff
@@ -1188,8 +1188,6 @@ bool city_has_building(const struct city *pcity,
                        const struct impr_type *pimprove)
 {
   if (nullptr == pimprove) {
-    /* Callers should ensure that any external data is tested with
-     * valid_improvement_by_number() */
```
Comment (lmoureaux) [inline common/city.cpp:1192]: This seems out of scope for unused code cleanup

Comment (pranavtifr) [reply common/city.cpp:1192]: Thats cuz, I deleted valid_improvement_by_number()
